plugins/woocommerce-blocks/tests/e2e/tests/mini-cart/mini-cart-block.merchant.block_theme.side_effects.spec.ts

```diff
@@ -54,15 +57,19 @@ test.describe( 'Merchant → Mini Cart', () => {
 				.getByLabel( 'Search for blocks and patterns' )
 				.fill( blockData.slug );
 
+			// Await for blocks commercial to be loaded in the Blocks inserter.
+			await expect(
+				editorUtils.page.locator(
+					'.block-directory-downloadable-block-list-item__details'
+				)
+			).toBeVisible();
```

Comment:
Can you help me to understand why this part is needed? I see that when commenting it out, that the test fails. I just don't have an explanation, currently, how this part is related to adding the Mini-Cart block only once. Are we using this as some kind of delay mechanism?

Reply:
When the promoted blocks load my impression was that the sidebar flashes. So my goal here was to wait for that area to be loaded so that I make sure to always select the same element.

Reply:
Thanks for clarifying this, @ralucaStan.